### ISSUE TITLE
Is seems to subscribe incorrect channel

### ISSUE THREAD

**Issue by takehilo** [socketio/socket.io-redis#162]
socket.io: v1.7.2
socket.io-redis: v2.0.1

Server:

```
const io = require('socket.io')();
const redis = require('socket.io-redis');

io.adapter(redis({host: 'localhost', port: 6379}));
io.listen(3000);
io.of('/chat');
```

Client: 

```
const socket = require('socket.io-client')('http://localhost:3000/chat');
```

I found that the server subscribes to incorrect channel:

```
$ redis-cli monitor
# Start Server
"subscribe" "socket.io#/#" "socket.io-request#/#" "socket.io-response#/#"
"subscribe" "socket.io#/chat#" "socket.io-request#/chat#" "socket.io-response#/chat#"

# Client connected
"subscribe" "socket.io#/#7OmQxtcBoa7h25o4AAAA#"
"subscribe" "socket.io#/chat#/chat#7OmQxtcBoa7h25o4AAAA#"
```

The final line shows that the server subscribes to the channel `socket.io#/chat#/chat#7OmQxtcBoa7h25o4AAAA#` but it should be `socket.io#/chat#7OmQxtcBoa7h25o4AAAA#`.

**Comment by darrachequesne** [comment]
Hi! Did you find your issue? Is there something to fix?

**Comment by takehilo** [comment]
At first, when I emitted an event using socket.io-emitter, I specified just the socket id like:

```
io.of('/chat').to('7OmQxtcBoa7h25o4AAAA').emit('chat', message);
```

this publishes the message to `socket.io#/chat#7OmQxtcBoa7h25o4AAAA#` but the channel is incorrect.

Later, I found that the correct socket ID was `chat#7OmQxtcBoa7h25o4AAAA` so I had to do the following:

```
io.of('/chat').to('chat#7OmQxtcBoa7h25o4AAAA').emit('chat', message);
```

It was just my misunderstanding. Thank you.
